### PR TITLE
DRAFT: Add support for filtering out metadata for libraries not sequenced yet

### PR DESCRIPTION
**Pull request by reisingerf**
Add `remove_not_sequenced` method to LabMetadataManager 
Add simple model tests for `get_by_keyword` and `get_by_keyword_in`

### CODE REVIEW

**Comment by reisingerf** [comment]
@victorskl that's what I got to so far. 
I think the view only uses the `get_by_keyword` method, so should be covered. 

Not sure about performance in real situations. And there is the question of the build in `get` method, which we'd probably have to overwrite if it should support the unsequenced filtering...

**Comment by victorskl** [comment]
Cool Flo; I will review this tomorrow.

File: data_portal/models/tests/test_labmetadata.py [hunk 1]
```diff
@@ -25,3 +26,49 @@ def test_get_labmetadata(self):
             self.assertEqual(get_complete_sequence.library_id, 'L2100001', 'Library result as expected.')
         except ObjectDoesNotExist:
             logger.info(f"Raised ObjectDoesNotExist which is not expected")
+
+    def test_get_by_keyword_not_sequenced(self):
+        # python manage.py test data_portal.models.tests.test_labmetadata.LabMetadataTestCase.test_get_by_keyword_not_requenced
```
Comment (victorskl) [inline data_portal/models/tests/test_labmetadata.py:31]: Typo in `requenced`. And also it is better in docstring wrap, pls.

Tip: In PyCharm, I use shortcut - cursor over function name; `Shift+Alt+CMD C`

```
python manage.py test data_portal.models.tests.test_labmetadata.LabMetadataTestCase.test_get_by_keyword_not_sequenced
```

File: data_portal/models/tests/test_labmetadata.py [hunk 2]
```diff
@@ -25,3 +26,49 @@ def test_get_labmetadata(self):
...
+        # The tumor library has no LibraryRun entry, i,e. has NOT been sequenced, therefore
+        # we expect to find it in a full metadata search
+        lib = LabMetadata.objects.get_by_keyword(library_id=TestConstant.library_id_tumor.value)
+        self.assertEqual(len(lib), 1, 'Expect matadata for tumor library')
+        # and NOT when excluding unsequenced libraries
+        lib = LabMetadata.objects.get_by_keyword(library_id=TestConstant.library_id_tumor.value, sequenced=True)
+        self.assertEqual(len(lib), 0, 'Did NOT expect metadat for tumor library (not sequenced yet)')
+
+    def test_get_by_keyword_in_not_sequenced(self):
+        # python manage.py test data_portal.models.tests.test_labmetadata.LabMetadataTestCase.test_get_by_keyword_in_not_requenced
```
Comment (victorskl) [inline data_portal/models/tests/test_labmetadata.py:54]: ditto typo `requenced`

File: data_portal/models/labmetadata.py [hunk 3]
```diff
@@ -85,11 +86,26 @@ class LabMetadataWorkflow(models.TextChoices):
     RESEARCH = "research"
 
 
+def remove_not_sequenced(qs: QuerySet) -> QuerySet:
```
Comment (victorskl) [inline data_portal/models/labmetadata.py:89]: Nice! 

1 function-_nality_ added; 2 unit test cases follow suit. Make things robust a tad more. That's the spirit, yes! 👍

**Review by victorskl — APPROVED** [review]
Looks great, Flo! Minor comment to attend.

I follow up user test and SQL cost in local. All good. Please see in thread. 
https://umccr.slack.com/archives/CP356DDCH/p1677557597056759

File: data_portal/models/tests/test_labmetadata.py (hunk 1 above)
Comment (reisingerf) [reply data_portal/models/tests/test_labmetadata.py]: Ah, sorry!
Yes I used PyCharm to get the function name, but then noticed the error in the actual function name and forgot to update the test call ;-)

Comment (reisingerf) [reply data_portal/models/tests/test_labmetadata.py]: Fixed.

File: data_portal/models/tests/test_labmetadata.py (hunk 2 above)
Comment (reisingerf) [reply data_portal/models/tests/test_labmetadata.py]: Fixed.

**Comment by reisingerf** [comment]
@victorskl  If you are happy I'll merge in a bit.
(noticed the thread and SQL cost comparison, nice one! And less than I expected)

**Pull request merged**